src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -1,255 +1,251 @@
             throws InvalidInputDataException, EntityNotFoundException {
         log.info("addProductToCart - process started");
-        Product product;
+        ProductDTO productDTO;
         try {
-            product = ProductMapper.DtoToProductEntity(productServiceClient.callGetProductByProductName(productName));
+            productDTO = productServiceClient.callGetProductByProductName(productName);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Product: " + productName + " not found in the Database!");
         }
 
-        if (product.getProductStock() == 0) {
+        if (productDTO.getStock() == 0) {
             throw new InvalidInputDataException("We are sorry, but currently: " + productName + " is out of order!");
         }
 
-        if (product.getProductStock() < quantity) {
-            throw new InvalidInputDataException("You can not add more than: " + product.getProductStock() + " "
+        if (productDTO.getStock() < quantity) {
+            throw new InvalidInputDataException("You can not add more than: " + productDTO.getStock() + " "
                     + productName + " Products to your shopping cart!");
         }
-        User user;
+        UserDTO user;
         try {
-            user = UserMapper.DtoToUserEntity(userServiceClient.callGetUserByEmail(email));
+            user = userServiceClient.callGetUserByEmail(email);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
-        Cart cart = cartRepository.findByUser(user);
+        Cart cart = cartRepository.findByUserId(user.getId());
 
-        List<Product> productList;
+        List<CartEntry> cartEntries;
         double productTotal = 0;
+
         if (cart == null) {
             cart = new Cart();
-            productList = new ArrayList<>();
+            cartEntries = new ArrayList<>();
+            CartEntry cartEntry = new CartEntry();
+            cartEntry.setProductName(productName);
+            cartEntry.setQuantity(quantity);
+            cartEntry.setCart(cart);
+
+            cartEntries.add(cartEntry);
         } else {
-            productList = cart.getProductList();
+            cartEntries = cart.getEntries();
+            CartEntry cartEntry = cartEntries.stream().filter(entry -> productName.equals(entry.getProductName())).findFirst().orElse(null);
+
+            if (cartEntry != null) {
+                cartEntry.setQuantity(cartEntry.getQuantity() + quantity);
+                cartEntry.setCart(cart);
+            }else {
+                CartEntry newCartEntry = new CartEntry();
+                newCartEntry.setProductName(productName);
+                newCartEntry.setQuantity(quantity);
+                newCartEntry.setCart(cart);
+                cartEntryRepository.save(newCartEntry);
```

Comment:
I think that this is not necessary because of the Cascade on the cartEntry

Reply:
the new cartEntry wount persist in the DB if I remove "cartEntryRepository.save(newCartEntry);"

Reply:
but you haven't add it to the cartEntries collection. it should persist it after you add it to the cartEntries(without calling the cartEntryRepository).

Reply:
you are right! done! 🥇